plugins/modules/rma_workflow_manager.py

```diff
@@ -1086,6 +1086,9 @@ def check_rma_task_status(self, task_id, success_message, error_prefix):
         while timeout_interval > 0:
```

Comment:
can we get progress and assign?

Why do we get progress again in line num 1097? Can't we optimize it?

```
while timeout_interval > 0:
    task_details = self.get_task_details(task_id)
    
    # Ensure 'response' and 'progress' exist in the task details
    progress = task_details.get('response', {}).get('progress', "").lower()
    
    # Handle error case
    if task_details.get("isError"):
        error_message = task_details.get("failureReason", "{0}: Task failed.".format(error_prefix))
        self.log(error_message, "ERROR")
        return {"status": "failed", "msg": error_message}
    
    # Check for successful completion
    if 'successful' in progress:
        self.log(success_message, "INFO")
        return {"status": "success", "msg": progress}
    
    # Log ongoing progress (optional)
    self.log("Task progress: {0}".format(progress), "DEBUG")
    
    # Wait and decrement the timeout interval
    time.sleep(ccc_poll_interval)
    timeout_interval -= ccc_poll_interval
```

Reply:
updated the code